lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,198 @@
+		target:     params.Target,
+		thumbprint: params.Thumbprint,
+	}
+
+	d, err := buildData(op, b, principal)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	d.ID = params.VchID
+
+	helper, err := getDatastoreHelper(op, d)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	output, err := getContentFromLogFiles(op, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	return operations.NewGetTargetTargetVchVchIDLogOK().WithPayload(output)
+}
+
+func (h *VCHDatacenterLogGet) Handle(params operations.GetTargetTargetDatacenterDatacenterVchVchIDLogParams, principal interface{}) middleware.Responder {
+	op := trace.NewOperation(params.HTTPRequest.Context(), "VCHDatacenterLogGet: %s", params.VchID)
+
+	b := buildDataParams{
+		target:     params.Target,
+		thumbprint: params.Thumbprint,
+		datacenter: &params.Datacenter,
+	}
+
+	d, err := buildData(op, b, principal)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	d.ID = params.VchID
+
+	helper, err := getDatastoreHelper(op, d)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	output, err := getContentFromLogFiles(op, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogOK().WithPayload(output)
+}
+
+// getDatastoreHelper validates the VCH and returns the datastore helper for the VCH. It errors when validation fails or when datastore is not ready
+func getDatastoreHelper(op trace.Operation, d *data.Data) (*datastore.Helper, error) {
+	// TODO (angiew): abstract some of the boilerplate into helpers in common.go
+	validator, err := validateTarget(op, d)
+	if err != nil {
+		return nil, util.WrapError(http.StatusBadRequest, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(http.StatusNotFound, fmt.Sprintf("Unable to find VCH %s: %s", d.ID, err))
+	}
+
+	if err := validate.SetDataFromVM(validator.Context, validator.Session.Finder, vch, d); err != nil {
```

Comment:
There is a `vch.FolderName` method that you could call that tells you the directory name of the VM.
It looks like that has been recently changed to only provide the Base name - I am unsure why and @caglar10ur has just left. However this is the call used in `lib/install/management/appliance.go:523`.

The ImageStores path MAY be the same, but should not be in the case of vSAN - a VM in vSAN MUST have a dedicated namespace and I believe we place the image VMDKS into their own namespace.

Reply:
Filled #6729 for this.